isic/studies/models.py

```diff
@@ -359,12 +379,35 @@ def view_annotation(user_obj, obj):
 class Response(TimeStampedModel):
     class Meta:
         unique_together = [['annotation', 'question']]
+        constraints = [
+            CheckConstraint(
+                name='response_choice_or_value_check',
+                check=Exact(
+                    lhs=Func(
+                        'choice',
+                        'value__value',
+                        function='num_nonnulls',
+                        output_field=IntegerField(),
+                    ),
+                    rhs=Value(1),
+                ),
+            )
+        ]
 
     annotation = models.ForeignKey(Annotation, on_delete=models.CASCADE, related_name='responses')
     question = models.ForeignKey(Question, on_delete=models.CASCADE, related_name='responses')
     # TODO: investigate limit_choices_to for admin capabilities
     # see: https://code.djangoproject.com/ticket/25306
-    choice = models.ForeignKey(QuestionChoice, on_delete=models.CASCADE)
+    choice = models.ForeignKey(QuestionChoice, on_delete=models.CASCADE, null=True)
+    # expect a single key inside named value. TODO: maybe add a constraint for this.
+    value = models.JSONField(default=dict)
```

Comment:
Rather than store an object with a `value` key, can we just directly store the value as a number type? Is there a reason to wrap everything in an object?